packages/esm-patient-notes-app/src/notes/visit-notes-form.component.tsx

```diff
@@ -154,24 +210,42 @@ const VisitNotesForm: React.FC<DefaultWorkspaceProps> = ({ closeWorkspace, patie
           },
         ],
         encounterType: encounterTypeUuid,
-        form: formConceptUuid,
         obs: obs,
       };
 
       const abortController = new AbortController();
       saveVisitNote(abortController, visitNotePayload)
         .then((response) => {
           if (response.status === 201) {
-            closeWorkspace();
-
-            showToast({
-              critical: true,
-              description: t('visitNoteNowVisible', 'It is now visible on the Encounters page'),
-              kind: 'success',
-              title: t('visitNoteSaved', 'Visit note saved'),
+            combinedDiagnosis.map((diagnosis, position: number) => {
+              let diagnosisPayload: DiagnosisPayload = {
+                encounter: response.data.uuid,
+                patient: patientUuid,
+                condition: null,
+                diagnosis: {
+                  coded: diagnosis.diagnosis.coded,
+                },
+                certainty: diagnosis.certainty,
+                rank: diagnosis.rank,
+              };
+              savePatientDiagnoses(abortController, diagnosisPayload).then((response) => {
+                if (response.status == 201) {
+                  if (position > -1) {
+                    combinedDiagnosis.splice(position, 1);
+                  }
+                  if (combinedDiagnosis.length == 0) {
+                    closeWorkspace();
+                    showToast({
+                      critical: true,
+                      description: t('visitNoteNowVisible', 'It is now visible on the Encounters page'),
+                      kind: 'success',
+                      title: t('visitNoteSaved', 'Visit note saved'),
+                    });
+                  }
+                  mutate(`/ws/rest/v1/encounter?patient=${patientUuid}&v=${encountersCustomRepresentation}`);
```

Comment:
It's weird to me that we keep all of this resetting flow inside the `savePatientDiagnoses()` call. We should really have a new `then()`  chained on to the end like this:

```ts
saveVisitNote(/*...*/)
  .then((response) => { 
    /* more code */
    return savePatientDiagnoses(/*...*/).then(/*...*);
  })
  .then(() => {
    closeWorkspace();
    showToast({
      critical: true,
      description: t('visitNoteNowVisible', 'It is now visible on the Encounters     page'),
      kind: 'success',
      title: t('visitNoteSaved', 'Visit note saved'),
    });
    mutate(`/ws/rest/v1/encounter?patient=${patientUuid}&v=${encountersCustomRepresentation}`);
  })
```